endpoint.go

```diff
@@ -767,13 +767,15 @@ func (ep *endpoint) assignAddress(assignIPv4, assignIPv6 bool) error {
 	}
 
 	if assignIPv4 {
-		if err = ep.assignAddressVersion(4, ipam); err != nil {
+		if err = ep.assignAddressVersion(4, ipam); err != nil && !ep.network.enableIPv6 {
 			return err
 		}
 	}
 
 	if assignIPv6 {
-		err = ep.assignAddressVersion(6, ipam)
+		if err = ep.assignAddressVersion(6, ipam); err != nil && ep.network.enableIPv6 {
```

Comment:
If we allow the auto-ipv6 allocation via capability, this check does not need to be modified.

Also, in case ipv6 is enabled and the user specifies an IPv6 subnet (no auto allocation), this change would hide a legitimate ipv6 address allocation failure, ehich is not correct.